### PR TITLE
OHRM5X-1828: Test buzz - comment API getAll, update, delete operations

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php [hunk 1]
```diff
@@ -386,18 +387,41 @@ public function getValidationRuleForDelete(): ParamRuleCollection
...
+     * @throws InvalidParamException
      */
     public function getOne(): EndpointResult
     {
-        throw $this->getNotImplementedException();
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $buzzPost = $this->getBuzzService()->getBuzzDao()->getBuzzPostById($id);
+        if (!$buzzPost instanceof BuzzPost) {
+            throw $this->getInvalidParamException(CommonParams::PARAMETER_ID);
+        }
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php:398]:
```suggestion
        $this->throwRecordNotFoundExceptionIfNotExist($buzzPost, BuzzPost::class);
```

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php [hunk 2]
```diff
@@ -386,18 +387,41 @@ public function getValidationRuleForDelete(): ParamRuleCollection
 
     /**
      * @inheritDoc
+     * @throws InvalidParamException
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php:390]:
```suggestion
```

File: src/plugins/orangehrmBuzzPlugin/test/Dao/BuzzDaoAdjustCountTest.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+use OrangeHRM\Buzz\Dao\BuzzDao;
+use OrangeHRM\Config\Config;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Entity\BuzzComment;
+use OrangeHRM\Entity\BuzzShare;
+use OrangeHRM\Tests\Util\KernelTestCase;
+use OrangeHRM\Tests\Util\TestDataService;
+
+class BuzzDaoAdjustCountTest extends KernelTestCase
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/test/Dao/BuzzDaoAdjustCountTest.php:30]: :ok_hand:

File: src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzCommentAPITestCases.yaml [hunk 4]
```diff
@@ -418,3 +418,473 @@ GetOne:
...
+        'lastName': 'Abel',
+        'firstName': 'Ashley',
+        'middleName': '',
+        'employeeId': '0005',
+        'terminationId': 1,
+      },
+    }
+    meta: [ ]
+
+  'Update Comment with large text (char count = 65530)':
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzCommentAPITestCases.yaml:725]: is this `65530` count correct?
if the char length equals 65530, this test case should pass

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzCommentAPITestCases.yaml (hunk 4 above)
Comment (ManulMax) [reply src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzCommentAPITestCases.yaml:725]: Previously checked with ENT version. Will check again

File: src/plugins/orangehrmBuzzPlugin/Service/BuzzService.php [hunk 5]
```diff
@@ -103,4 +104,18 @@ public function canDeleteBuzzComment(int $commentOwnerEmpNumber): bool
         }
         return $this->buzzCommentPermissionCache[$self]->canDelete();
     }
+
+    /**
+     * @param int $postOwnerEmpNumber
+     * @return bool
+     */
+    public function canUpdateBuzzPost(int $postOwnerEmpNumber): bool
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Service/BuzzService.php:112]: can't we reuse `canUpdateBuzzFeedPost`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzCommentAPITestCases.yaml (hunk 4 above)
Comment (ManulMax) [reply src/plugins/orangehrmBuzzPlugin/test/fixtures/testcases/BuzzCommentAPITestCases.yaml:725]: Checked, char length here is 65531

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php [hunk 6]
```diff
@@ -550,16 +576,11 @@ public function update(): EndpointResult
...
-                $buzzFeedFilterParams->setAuthUserEmpNumber($this->getAuthUser()->getEmpNumber());
-                $buzzFeedFilterParams->setShareId($buzzShare->getId());
-                $buzzFeedPosts = $this->getBuzzService()->getBuzzDao()->getBuzzFeedPosts($buzzFeedFilterParams);
-                $buzzPost = $buzzFeedPosts[0];
+                $buzzPost = $this->getBuzzFeedPost($postId);
             }
 
             $this->commitTransaction();
-        } catch (InvalidParamException|BadRequestException $e) {
+        } catch (InvalidParamException|ForbiddenException $e) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php:583]: :ok_hand:

File: src/plugins/orangehrmBuzzPlugin/Service/BuzzService.php [hunk 7]
```diff
@@ -31,6 +31,7 @@ class BuzzService
     private BuzzLikeDao $buzzLikeDao;
     private array $buzzFeedPostPermissionCache = [];
     private array $buzzCommentPermissionCache = [];
+    private array $buzzPostPermissionCache = [];
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/Service/BuzzService.php:34]: remove this line

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**